01_class_change_page/class-give-page.php

```diff
@@ -0,0 +1,129 @@
+<?php
+ini_set("display_errors", 0);
+error_reporting(E_ALL);
+require_once('submit.php');
+
+// DB接続
+try {
+    $pdo = new PDO($dsn, $user, $password);
+    $pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+} catch (Exception $e) {

```

Comment:
[nits]
PDOに失敗したときはPDOException（RuntimeExceptionのサブクラス）の例外が投げられるので、例外設計としては下記のように記述するのが適切かと思います！

```suggestion
} catch (PDOException $e) {

```

この記述は、「データベース接続に失敗したときに例外をcatchして処理をしたい」という意図があるかと思います。
しかし、Exceptionにしてしまうとデータベース接続エラー以外の想定外のエラーもキャッチしてしまうため、意図した挙動にならない可能性があります。
それとエラーログに「Exception~」と出ているより、「PDOException~」と出ていれば「あ、これはデータベース接続に失敗しているんだな」とすぐわかります。

適切な例外設計をすることで障害があったときの原因対応もやりやすくなるので、この辺も少しずつ意識できるとよりGoodです 👍